test/test_job.py

```diff
@@ -14,6 +14,7 @@
 # pylint: disable=protected-access
 
 import unittest
+from unittest import mock
```

Comment:
Is this needed?

Reply:
The test failed locally, so i added this.
If it doesn't fail on GitHub and/or your system, i can removed it.

Reply:
It's going to block CI, the lint rules block unused imports and nothing is using `mock` plain (it's only used as `unittest.mock`)

Reply:
You're right. I removed it and ran `tox`, so hopefully the CI goes green.